src/mozanalysis/frequentist_stats/sample_size.py

```diff
@@ -0,0 +1,313 @@
+
+    Args:
+        res: A TimeSeriesResult, generated by
+            mozanalysis.sizing.HistoricalTarget.get_time_series_data.
+        bq_context: A mozanalysis.bq.BigQueryContext object that handles downloading
+            time series data from BigQuery.
+        metrics_list (list of mozanalysis.metrics.Metric): List of metrics
+            used to construct the results df from HistoricalTarget. The names
+            of these metrics are used to return results for sample size
+            calculation for each.
+        quantile (float, default .90): Quantile used to calculate the effect size
+            as the quantile of week-to-week metric changes and the variance of
+            the mean.
+        alpha (float, default .05): Significance level for the experiment.
+        power (float, default .90): Probability of detecting an effect,
+            when a significant effect exists.
+        parent_distribution (str, default "normal"): Distribution of the parent data;
+            must be normal, uniform, logistic, or laplace.
+
+    Returns a sorted dictionary:
+        Keys in the dictionary are the metrics column names from the DataFrame; values
+        are the required sample size to achieve the desired power for that metric.
+        The dictionary is sorted by required sample size, in descending order.
+    """
+
+    def _mann_whitney_solve_sample_size_approximation(
+        effect_size,
+        std,
+        alpha=.05,
+        power=.8,
+        parent_distribution="normal"
+    ):
+
+        """
+        Wilcoxen-Mann-Whitney rank sum test sample size calculation,
+        based on asymptotic efficiency relative to the t-test.
+        """
+        rel_effect_size = effect_size/std
+        are = {
+            "uniform": 1.,
+            "normal": pi/3.,
+            "logistic": 9./(pi**2),
+            "laplace": 2./3.
+        }
+
+        if parent_distribution not in are.keys():
+            raise ValueError(f"Parent distribution must be in {are.keys()}")
+
+        t_sample_size = tt_ind_solve_power(
+            effect_size=rel_effect_size,
+            power=power,
+            alpha=alpha
+        )
+
+        return t_sample_size*are[parent_distribution]
+
+    res_mean, pop_size = res.get_aggregated_data(
+        bq_context=bq_context,
+        metric_list=metric_list,
+        aggregate_function="AVG"
+    )
+
+    res_mean.sort_values(by="analysis_window_start", ascending=True, inplace=True)
+    effect_size = {
+        m.name: res_mean[m.name].diff().abs().quantile(q=quantile) for m in metric_list
+        }
+
+    res_std, _ = res.get_aggregated_data(
+        bq_context=bq_context,
+        metric_list=metric_list,
+        aggregate_function="STDDEV"
+    )
+    std = {m.name: res_std[m.name].quantile(q=quantile) for m in metric_list}
+
+    size_dict = {
+        m.name: {
+            "effect_size": effect_size[m.name],
+            "std_dev": std[m.name],
```

Comment:
@m-d-bowerman could we add to this output the analysis windows that the effect size and standard deviation correspond to?

Reply:
Just to make sure I understand, you want it to return the time period that corresponds to the 90th percentile for each statistic?

Reply:
yes - either the exact dates of the time period, or even a window identifier would be fine.

Reply:
The change to add the analysis window to the results dictionary is included in the latest commit

Reply:
thank you!